app/controllers/util/DynamicFormUtils.scala

```diff
@@ -16,38 +16,49 @@ class DynamicFormUtils(request: Request[AnyContent], defaultFieldValues: List[Fo
 
   def validateAndConvertSubmittedValuesToFormFields(submittedValues: Map[String, Seq[String]]): List[FormField] = {
```

Comment:
Maybe rename this to `convertSubmittedValuesToFormFields` The validation bit is implicit from calling the validation method.